docs/getting_started/tutorial.rst

```diff
@@ -0,0 +1,376 @@
+:doc:`docs <../topics/images/index>`.
+
+Blog Index
+~~~~~~~~~~
+
+Let us extend the Blog app to provide an index.
+
+.. code:: python
+
+    class BlogIndexPage(Page):
+        intro = RichTextField(blank=True)
+
+        content_panels = Page.content_panels + [
+            FieldPanel('intro', classname="full")
+        ]
+
+The above creates an index type to collect all our blog posts.
+
+``blog/templates/blog/blog_index_page.html``
+
+.. code:: html+django
+
+    {% extends "base.html" %}
+
+    {% load static core_tags wagtailcore_tags %}
+
+    {% block body_class %}template-{{ self.get_verbose_name|slugify }}{% endblock %}
+
+    {% block content %}
+        <h1>{{ self.title }}</h1>
+
+        <div class="intro">{{ self.intro }}</div>
+    {% endblock %}
+
+Related items
+~~~~~~~~~~~~~
+
+Let's extend the BlogIndexPage to add related links. The related links
+can be BlogPages or external links. Change ``blog/models.py`` to
+
+.. code:: python
+
+    from django.db import models
+
+    from modelcluster.fields import ParentalKey
+
+    from wagtail.wagtailcore.models import Page, Orderable
+    from wagtail.wagtailcore.fields import RichTextField
+    from wagtail.wagtailadmin.edit_handlers import (FieldPanel,
+                                                    InlinePanel,
+                                                    MultiFieldPanel,
+                                                    PageChooserPanel)
+    from wagtail.wagtailimages.edit_handlers import ImageChooserPanel
+    from wagtail.wagtailsearch import index
+
+
+    class BlogPage(Page):
+        main_image = models.ForeignKey(
+            'wagtailimages.Image',
+            null=True, blank=True,
+            on_delete=models.SET_NULL,
+            related_name='+'
+        )
+        date = models.DateField("Post date")
+        intro = models.CharField(max_length=250)
+        body = RichTextField(blank=True)
+
+        search_fields = Page.search_fields + (
+            index.SearchField('intro'),
+            index.SearchField('body'),
+        )
+
+        content_panels = Page.content_panels + [
+            FieldPanel('date'),
+            ImageChooserPanel('main_image'),
+            FieldPanel('intro'),
+            FieldPanel('body'),
+        ]
+
+
+    class LinkFields(models.Model):
+        link_external = models.URLField("External link", blank=True)
+        link_page = models.ForeignKey(
+            'wagtailcore.Page',
+            null=True,
+            blank=True,
+            related_name='+'
+        )
+
+        @property
+        def link(self):
+            if self.link_page:
+                return self.link_page.url
+            else:
+                return self.link_external
+
+        panels = [
+            FieldPanel('link_external'),
+            PageChooserPanel('link_page'),
+        ]
+
+        class Meta:
+            abstract = True
+
+
+    # Related links
+
+    class RelatedLink(LinkFields):
```

Comment:
I could cut out the `BlogPage` definition as that doesn't change. Not sure what else should be cut